### PR TITLE
Sprint 1/issue 1# create sign up screen

### LINKED ISSUE #1: Create Sign up Screen
# REQUIREMENTS
Create a registration screen where users can sign up with their name, email, and password. The registration must be secure to not compromise the users' information.
# DESIGN
 How it should look like.
signup

# Acceptance Criteria
List all required acceptance criteria
- [x] Sign Up Layout.
- [x] Conect with firebase for the SignUp
- [x] Validations for the inputs

### PR DESCRIPTION
**Pull request by ernesto2001-code**
# Description

- We have added several components and functionalities to enhance the user experience in the registration process. This includes creating InputText, Button, and Checkbox components, as well as implementing state logic to handle dynamic styling and component interactivity.
- Additionally, we have integrated Firebase with authentication to ensure a secure and efficient sign-up process.
With these changes, users can interact more effectively with the registration interface, improving the user experience and facilitating the registration process.

## Screenshots

signUp

### CODE REVIEW

File: src/components/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+//     flexDirection: 'row',
+//     marginBottom: 20,
+//   },
+//   checkbox: {
+//     backgroundColor: '#5C6EF8',
+//   },
+//   label: {
+//     margin: 8,
+//   },
+// });
```
Comment (juperameza) [inline src/components/Checkbox.tsx]: If this code isn't needed delete it

File: src/navigator/StackNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export const StackNavigator = () => {
+  return (
+    <Stack.Navigator
+        screenOptions={{ headerShown: false }}
+    >
+      <Stack.Screen name="SignUp" component={ SignUp } />
+      <Stack.Screen name="HomePageScreen" component={ HomePageScreen } />
+    </Stack.Navigator>
+  );
+}
```
Comment (juperameza) [inline src/navigator/StackNavigator.tsx]: Add a new line at the end of this file

File: src/screens/HomePageScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  
+    return (
+      <View>
+        <Text>Welcome {user.email}</Text>
+        <Button title="Log Out" onPress={handleLogOut} />
+      </View>
+    );
+  }
+
+  export default HomePageScreen;
```
Comment (juperameza) [inline src/screens/HomePageScreen.tsx]: Same here new line at the end of this file

File: src/screens/SignUpScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  const emailInvalid = (reason: string) => {
+    setEmailValid(reason);
+  }
+
+  const SignUpTest = () =>{
+    auth()
+      .createUserWithEmailAndPassword( email, password)
+      .then((userCredential) => {
+        const user = userCredential.user;
+        // Alert.alert("Usuario Creado! " + email)
```
Comment (juperameza) [inline src/screens/SignUpScreen.tsx]: Same here if the code isn't needed anymore delete It

**Review by juperameza — CHANGES_REQUESTED** [review]
Here's why I leave those comments
[Lines at the end of the file](https://gist.github.com/OleksiyRudenko/d51388345ea55767b7672307fe35adf3)
[Commented code](https://methodpoet.com/commented-out-code/)

**Review by juperameza — APPROVED** [review]

**Comment by AxeLoza21** [comment]
I agree with juperameza, if you don't use the commented code, delete it. For the rest I see it fine

File: src/components/Checkbox.tsx (hunk 1 above)
Comment (ernesto2001-code) [reply src/components/Checkbox.tsx]: thanks!, fixed.

**Pull request merged**